### PR TITLE
new pkg: sonar-scanner-cli

### PR DESCRIPTION
**Pull request by vanlongme**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [x] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: sonar-scanner-cli.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package:
+  name: sonar-scanner-cli
+  version: 5.0.1.3006
+  epoch: 0
+  description: Scanner CLI for SonarQube and SonarCloud
+  copyright:
+    - license: GNU Lesser General Public License v3.0
```
Comment (ajayk) [inline sonar-scanner-cli.yaml:7]: Needs to be valid short [SPDX Identifier ](https://spdx.org/licenses/LGPL-3.0-or-later.html)
```suggestion
    - license: LGPL-3.0-or-later
```




```

File: sonar-scanner-cli.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      mvn clean package
+      mkdir -p ${{targets.destdir}}/usr/bin/sonarqube
+      cp target/sonar-scanner-cli-*.jar ${{targets.destdir}}/usr/bin/sonarqube/sonarscanner-cli.jar
+      chmod +x sonar-scanner-cli && cp sonar-scanner-cli ${{targets.destdir}}/usr/bin/
+
+update:
+  enabled: true
+  github:
+    identifier: sonarqube/sonar-scanner-cli
+    use-tag: true
```
Comment (rawlingsj) [inline sonar-scanner-cli.yaml:44]:
```suggestion
    use-tag: true

test:
  pipeline:
    - runs: |
        sonar-scanner --version
```

File: sonar-scanner-cli.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package:
+  name: sonar-scanner-cli
+  version: 5.0.1.3006
+  epoch: 1
```
Comment (ajayk) [inline sonar-scanner-cli.yaml:4]:
```suggestion
  epoch: 0
```

File: sonar-scanner-cli.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  description: Scanner CLI for SonarQube and SonarCloud
+  copyright:
+    - license: LGPL-3.0-or-later
+  dependencies:
+    runtime:
+      - bash
+      - openjdk-17-default-jvm
+      - openjdk-17-jre
```
Comment (ajayk) [inline sonar-scanner-cli.yaml:12]: dropping this would allow the package to run in any JRE >=17 
```suggestion
```

File: sonar-scanner-cli.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      cp target/sonar-scanner-cli-*.jar ${{targets.destdir}}/usr/bin/sonarqube/sonarscanner-cli.jar
+      chmod +x sonar-scanner-cli && cp sonar-scanner-cli ${{targets.destdir}}/usr/bin/
+
+update:
+  enabled: true
+  github:
+    identifier: sonarqube/sonar-scanner-cli
+    use-tag: true
+
+test:
```
Comment (ajayk) [inline sonar-scanner-cli.yaml:49]: you can check the [maven](https://github.com/wolfi-dev/os/blob/main/maven.yaml#L65-L72) test example 
```suggestion
test:
  environment:
    contents:
      packages:
        - openjdk-17-default-jvm
        - openjdk-17
```

Comment (ajayk) [reply sonar-scanner-cli.yaml:49]: This is from a test I ran locally 
```
export PATH='/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin'

[ -d '/home/build' ] || mkdir -p '/home/build'
cd '/home/build'
sonar-scanner-cli --version

exit 0
2024/03/21 06:05:52 INFO INFO: Scanner configuration file: NONE
2024/03/21 06:05:52 INFO INFO: Project root configuration file: NONE
2024/03/21 06:05:52 INFO INFO: SonarScanner 5.0.1-SNAPSHOT
2024/03/21 06:05:52 INFO INFO: Java 17.0.10-internal wolfi (64-bit)
2024/03/21 06:05:52 INFO INFO: Linux 5.15.120+ amd64

```

Comment (vanlongme) [reply sonar-scanner-cli.yaml:49]: thanks, LGTM

**Review by ajayk — APPROVED** [review]

**Pull request merged**